src/index.ts

```diff
@@ -20,16 +19,21 @@ const service = () => {
       token: token,
       mesgConfig: mesgConfig,
       client: new ClientBuilder({
-        endpoint: endpointTCP,
+        endpoint: defaultEndpoint,
       }).service(),
     });
   }
 
   return defaultService;
 }
 
-const application = () => {
-  if(!defaultApplication){
+type ApplicationOptions = {
+  endpoint?: string
+}
+
+const application = (options?: ApplicationOptions) => {
```

Comment:
Could you move service and application functions to `service/index.ts` and `application/index.ts`?

It's better if we can separate as much service and application.

Reply:
Except the import they don't have anything in common.

For endpoint, only in service should read `process.env.MESG_ENDPOINT`. In application, it should get one from the user or fallback to the default hardcoded `localhost:50052`.

Reply:
Can you open separate issues for these? I already made some other changes in this file for an another PR.